CMakeLists.txt

```diff
@@ -45,6 +46,17 @@ if(DEFINED ENV{VCPKG_DEFAULT_TRIPLET} AND NOT DEFINED VCPKG_TARGET_TRIPLET)
 endif()
 
 project(az LANGUAGES C)
+
+if (MSVC)
+    if (STATIC_CRT)
+        set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /MT")
+        set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} /MTd")
```

Comment:
Do we always need to set for both flavors of the build?

Unrelated, but maybe we should have a simplified Debug and Release option (see https://github.com/Azure/azure-sdk-for-c/issues/639)

Reply:
I think we should, so it is guaranteed to be predictable, and not the same thing, even if defaults change.

Reply:
Regarding setting both - what I like is that the correct option is picked up, depending on what release flavor is chosen.
We can add release/debug flag, but I think it should be done as a separate PR.